lightning/src/ln/channelmanager.rs

```diff
@@ -18220,6 +18242,35 @@ where
 			}
 		}
 
+		// In the future, the set of pending HTLCs will be pulled from `Channel{Monitor}` data and
+		// placed in `ChannelManager::decode_update_add_htlcs` on read, to be handled on the next call
+		// to `process_pending_htlc_forwards`. This is part of a larger effort to remove the requirement
+		// of regularly persisting the `ChannelManager`. However, this new pipeline cannot currently
+		// handle inbound HTLC receives, some of which may be present in `failed_htlcs`, so continue
+		// handling HTLCs present in `failed_htlcs` during deserialization for now.
```

Comment:
It sounds like the code below isn't necessary yet, but a preparation for the future. Is that right?